### DISCUSSION TITLE
JSON API

### DISCUSSION THREAD

**Discussion by michalsiman** [magicbug/Cloudlog#2751] [General]
It stil JSON API for QSO working? I am playing with python and requests, but nothing happend ... btw station_profile_id ...

`
url = 'https://mycloudlogurl/'
data = {'key':'myvalidapikey',
        'station_profile_id':2,
        'type':'adif',
        'string':'<call:5>TS1TST<band:4>20m<mode:3>SSB<freq:10>14.150000<qso_date:8>20231129<time_on:6>130000<time_off:6>130000<rst_rcvd:2>59<rst_sent:2>55<eor>'}
`

**Comment by phl0** [comment]
Can you have a log at your Cloudlog to see if it contains any errors? Maybe increase the logging threshold to 4 in application/config/config.php beforehand.

**Reply by michalsiman** [reply]
where is the log where i can see api reaction? btw response answer is 200 ...

**Comment by phl0** [comment]
Btw the ADIF length information is incorrect in your example:

```
<band:4>20m should be <band:3>20m and 
<freq:10>14.150000 should be <freq:9>14.150000
```

**Reply by michalsiman** [reply]
correct, sorry, that is my fault, but still nothing happend ...

**Comment by magicbug** [comment]
Think @phl0 has already pointed it out but your ADIF string is full of errors.

**Reply by michalsiman** [reply]
i use adif string from example from wiki, yes, lenght of frekvency field is wrong, but what else?

**Reply by phl0** [reply]
length of band.

**Reply by phl0** [reply]
The wiki is correct. You just replaced the band without adjusting the length of the ADIF tag. This will fail.

**Reply by michalsiman** [reply]
ok ok, sorry, yes, all lenght, i am correct it now, but still nothing happend ...

`'string':'<call:6>OK9TST<band:3>20m<mode:3>SSB<freq:9>14.150000<qso_date:8>20231129<time_on:6>130000<time_off:6>130000<rst_rcvd:2>59<rst_sent:2>55<eor>'`

**Reply by phl0** [reply]
Then set the logging threshold to 4 and have a log at the log file created in application/logs.

**Reply by michalsiman** [reply]
ok, thx

**Comment by michalsiman** [comment]
and what is station_progile_id? i have no station_profile in cloudlog anymore ...

**Reply by phl0** [reply]
That is the number of your station location.

**Reply by michalsiman** [reply]
ok, thx
